app/vmctl/influx/parser.go

```diff
@@ -36,6 +36,75 @@ func parseResult(r influx.Result) ([]queryValues, error) {
 	return qValues, nil
 }
 
+// parseResultCheckTags check rows return by InfluxDB and remove rows not belong to current series.
+// See: https://github.com/VictoriaMetrics/VictoriaMetrics/issues/7301.
+func parseResultCheckTags(s *Series, r influx.Result) ([]queryValues, error) {
+	if len(r.Err) > 0 {
+		return nil, fmt.Errorf("result error: %s", r.Err)
+	}
+
+	// This map contains all the desired column of the query result for current series.
+	// If a column (`Columns[i]` in influx.Result) not exists in map, this column must be declared by other series.
+	// A row (`values[i]` in influx.Result) is considered invalid if its unwanted column value is not null.
+	wantedColumns := make(map[string]bool, len(s.LabelPairs)+len(s.FieldList)+1)
+	for i := range s.LabelPairs {
+		wantedColumns[s.LabelPairs[i].Name] = true
+	}
+	// Multiple fields might exist, and they should be allowed.
+	for i := range s.FieldList {
+		wantedColumns[s.FieldList[i]] = true
+	}
+	wantedColumns["time"] = true // const from InfluxDB
+
+	for i := range r.Series {
+		if len(s.LabelPairs)+2 > len(r.Series[i].Columns) {
+			// column in series fewer than query where condition. should never reach
+			return nil, fmt.Errorf(`wrong number of columns in result series, expected: %v, "%s" and "time", got %v`, s.LabelPairs, s.Field, r.Series[i].Columns)
+		}
+		// prepare a new values slice to replace the existing one.
+		values := make([][]interface{}, 0, len(r.Series[i].Values))
+
+		// prepare an unwanted column list
+		unwantedColumnIdx := make([]int, 0)
+		for idx := range r.Series[i].Columns {
+			if !wantedColumns[r.Series[i].Columns[idx]] {
+				unwantedColumnIdx = append(unwantedColumnIdx, idx)
+			}
+		}
+
+		// go through each rows
+		for j := range r.Series[i].Values {
+			if len(r.Series[i].Columns) != len(r.Series[i].Values[j]) {
+				// column in a row does not match series columns. should never reach
+				return nil, fmt.Errorf(`wrong number of columns in result row, expected: %d, got: %d`, len(r.Series[i].Columns), len(r.Series[i].Values[j]))
+			}
+
+			// skip if value of unwanted column is not null.
+			skip := false
```

Comment:
i think we can remove `skip` variable and just use 
```golang
for _, idx := range unwantedColumnIdx {
    if r.Series[i].Values[j][idx] != nil {
	continue
    }
    values = append(values, r.Series[i].Values[j])
}

Reply:
If unwanted column is not `nil`, we need to `continue` on the outer `for-loop`, not the current one. I think `goto` is needed here if we want to remove `skip` flag. WDYT?

Reply:
It depends on how each person likes to write code. :)